tasks/add-unique-granuleID/src/types.ts

```diff
@@ -0,0 +1,23 @@
+// Either collectionId OR (datatype AND version) must be present
+export type GranuleInput =
+  | (BaseGranule & { collectionId: string; dataType: never; version: never })
+  | (BaseGranule & { collectionId: never; dataType: string; version: string });
+
+interface BaseGranule {
+  granuleId: string;
+  producerGranuleId?: string;
+}
+
+export type HandlerInput = {
+  granules: GranuleInput[],
+};
+
+export type HandlerOutput = {
```

Comment:
The output would always have `producerGranuleId` I think. Maybe we should have a different type that asserts that's true?

Reply:
Hmm, yes, excellent point.    

I went ahead and did that, mildly abusing an assertion - which given the explicit mutation and TS limits on that seemed appropriate in this case, particularly given the output typing likely won't be used for anything other than clarity. 

Let me know what you think: f3db6d1ff25

Reply:
Looks good! Agreed that this is really just for clarity. Just nice to be able to look at the type and see immediately what we're expecting.